derive-macros/src/lib.rs

```diff
@@ -107,6 +107,64 @@ fn gen_schema_fields(data: &Data) -> TokenStream {
                 _ => false,
             }
         });
+        // Validate field_id attribute and collect any errors
+        let mut field_id_errors = Vec::new();
+        let _field_id: Option<i64> = field.attrs.iter().find_map(|attr| {
+            match &attr.meta {
+                Meta::NameValue(nv) if nv.path.get_ident().is_some_and(|ident| ident == "field_id") => {
+                    match &nv.value {
+                        syn::Expr::Lit(expr_lit) => {
+                            match &expr_lit.lit {
+                                Lit::Int(lit_int) => {
+                                    match lit_int.base10_parse() {
+                                        Ok(value) => Some(value),
+                                        Err(e) => {
+                                            field_id_errors.push(Error::new(
+                                                lit_int.span(),
+                                                format!("Invalid field_id value: failed to parse integer: {}", e)
+                                            ));
+                                            None
+                                        }
+                                    }
+                                }
+                                Lit::Str(lit_str) => {
+                                    field_id_errors.push(Error::new(
+                                        lit_str.span(),
+                                        "field_id must be an integer literal, not a string"
+                                    ));
+                                    None
+                                }
+                                _ => {
+                                    field_id_errors.push(Error::new(
+                                        expr_lit.span(),
+                                        "field_id must be an integer literal"
+                                    ));
+                                    None
+                                }
+                            }
+                        }
+                        _ => {
+                            field_id_errors.push(Error::new(
+                                nv.value.span(),
+                                "field_id must be an integer literal"
+                            ));
+                            None
+                        }
+                    }
+                }
+                _ => None,
+            }
+        });
+
+        // If we found validation errors, return them instead of proceeding
+        if !field_id_errors.is_empty() {
+            let mut errors_iter = field_id_errors.into_iter();
+            let mut combined_error = errors_iter.next().unwrap();
```

Comment:
Slightly simpler:
```suggestion
        let mut errors_iter = field_id_errors.into_iter();
        if let Some(mut combined_error) = errors_iter.next() {
```
(but honestly this whole thing seems like big overkill, to collect and format multiple errors?)